### PR TITLE
Project Day 2: Transactions and queries

### PR DESCRIPTION
**Pull request by jonas-45**
<h3>In this milestone, SQL transactions, and aggregates were implemented in queries. The implementations are given below;</h3>

- [ ] Data was added to the animal's table
Screenshot2

- [ ] Inside a transaction, the animal's table was updated by setting the species column to unspecified and then rolling back the change.
Screenshot1

- [ ] Inside a transaction, the animal's table was updated by setting the species column to digimon for all animals that have a name ending in 'mon' and also setting species to pokemon for all animals that don't have species already set.
Screenshot3

- [ ] All the data in the table was deleted and rolled back in a transaction
Screenshot4

- [ ] In a transaction, specific animals were deleted, created a savepoint, updated the animals table and rolled back to the savepoint
Screenshot5

<h3>The following questions were answered with queries</h3>

- [ ] How many animals are there?
Screenshot6

- [ ] How many animals have never tried to escape?
Screenshot7

- [ ] What is the average weight of animals?
Screenshot8

- [ ] Who escapes the most, neutered or not neutered animals?
Screenshot9

- [ ] What is the minimum and maximum weight of each type of animal?
Screenshot10

- [ ] What is the average number of escape attempts per animal type of those born between 1990 and 2000?
Screenshot11

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -14,4 +14,42 @@ SELECT * FROM animals WHERE neutered = true
...
+
+/* Project DAY 2 queries */
+BEGIN;
+UPDATE animals SET species = 'unspecified';
+ROLLBACK;
+
+BEGIN;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+COMMIT;
```
Comment (brytebee) [inline queries.sql]: For this transaction and others, after every activity you are required to verify the activity worked and when there is a rollback or commit to also verify.

image

All verification should be made as shown below.

SELECT * FROM animals;
(select all from the specified table in the case above the animals table.)

File: queries.sql [hunk 2]
```diff
@@ -14,4 +14,42 @@ SELECT * FROM animals WHERE neutered = true
...
+
+BEGIN;
+DELETE FROM animals;
+ROLLBACK;
```
Comment (brytebee) [inline queries.sql:35]: Here.

File: queries.sql [hunk 3]
```diff
@@ -14,4 +14,42 @@ SELECT * FROM animals WHERE neutered = true
...
+
+BEGIN;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT SP1;
+UPDATE animals SET weight_kg = weight_kg * (-1);
+ROLLBACK TO SP1;
+UPDATE animals SET weight_kg = weight_kg * (-1) WHERE weight_kg < 0;
+COMMIT;
```
Comment (brytebee) [inline queries.sql:46]: Here too. Kindly look through the instruction for other verify points.

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi, @jonas-45👋,

##

### Highlights💡
- Good commit message.✔️ 
- Good PR Summary.✔️
- Great DB design.✔️
- App functions as it should.✔️

##

Good job so far coming so far with the progress you've made! 💪💪
 
However, there are some issues that you still need to work on to go to the next project but you are almost there!



# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Please, remember to tag me in your question so I can receive the notification.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 3 above)
Comment (OyePriscilla) [reply queries.sql:46]: - [ ] Please as raised earlier, kindly verify (**as you rightly did for others**) that after the `ROLLBACK` in line 42 and `COMMIT` in line 44 that all records in animal table still exist. 

Untitled

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
Hi @jonas-45 👏 👏 , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

so-close-so-close-seth-meyers

 ## Highlights
Use database transactions ✔️ 
Modify and delete data in SQL ✔️ 
Create queries to answer some questions in transactions ✔️ 
Made some of the required changes ✔️ 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by jonas-45** [comment]
@OyePriscilla thanks for your time and effort in reviewing my work. Please I implemented all the changes requested by @brytebee. What you are referring to was corrected in line 31. 
Screenshotv

We are not required to verify change in line 42 and 44;
Screenshot 2023-04-05 004732

**Review by Reem-lab — APPROVED** [review]
# Hi @jonas-45,👋🏻👋🏻

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🔥🔥💯💯


## Highlights 🔥🔥🔥🔥
- Followed Gitflow ✅
- Great working on the functionality 🔥🔥




## Optional suggestions



_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**